src/Disks/ObjectStorages/HDFS/HDFSObjectStorage.cpp

```diff
@@ -101,18 +101,6 @@ std::unique_ptr<WriteBufferFromFileBase> HDFSObjectStorage::writeObject( /// NOL
 }
 
 
-void HDFSObjectStorage::listPrefix(const std::string & path, RelativePathsWithSize & children) const
-{
-    const size_t begin_of_path = path.find('/', path.find("//") + 2);
-    int32_t num_entries;
-    auto * files_list = hdfsListDirectory(hdfs_fs.get(), path.substr(begin_of_path).c_str(), &num_entries);
-    if (num_entries == -1)
-        throw Exception(ErrorCodes::HDFS_ERROR, "HDFSDelete failed with path: " + path);
-
-    for (int32_t i = 0; i < num_entries; ++i)
-        children.emplace_back(files_list[i].mName, files_list[i].mSize);
-}
```

Comment:
Was this implementation wrong?

Reply:
Yes, since this is analog of `ls -l` AFAIR, and it is not recursive.

Reply:
So you've deleted it for now but in order to implement `hdfs_plain` we'll need to add it back, right? In this case I suppose it would be better to keep the old implementation but probable comment it with `#if 0` and a comment.

Reply:
>So you've deleted it for now but in order to implement hdfs_plain we'll need to add it back, right? In this case I suppose it would be better to keep the old implementation but probable comment it with #if 0 and a comment.

For it we need both, `findAllFiles` and `getDirectoryContents`, so it should be reimplemented anyway.